bin/oci-notify

```diff
@@ -0,0 +1,404 @@
+    log "Publishing message '$header'"
+    #
+    # oci cli profile
+    if [ "$instance_principal" -eq 0 ]; then
+        log_debug "Running OCI CLI with profile config."
+        out0=$(oci $OCI_CLI_PROFILE_CONFIG ons message publish \
+                  --topic-id "$topicid"  --title "$header" --body "$body" 2>&1)
+        ret0=$?
+        log_debug "$out0"
+        if [ $ret0 -ne 0 ]; then
+            #
+            # wait a second, we do second try with instance principal auth
+            sleep 1
+            log_debug "Running OCI CLI with instance principal auth."
+            out1=$(oci $OCI_INSTANCE_PRINCIPAL_AUTH ons message publish \
+                --topic-id "$topicid" --title "$header" --body "$body" 2>&1)
+            ret1=$?
+            if [ $ret1 -ne 0 ]; then
+                log_debug "$out1"
+                log_error "Failed to publish message '$header'" >&2
+                return $ret0
+            else
+                log "Published message '$header'"
+            fi
+        else
+            log "Published message '$header'"
+        fi
+    else
+        # instance principal auth
+        log_debug "Running OCI CLI with instance principal auth."
+        out1=$(oci $OCI_INSTANCE_PRINCIPAL_AUTH ons message publish \
+            --topic-id "$topicid" --title "$header" --body "$body" 2>&1)
+        ret1=$?
+        log_debug "$out1"
+        if [ $ret1 -ne 0 ]; then
+            log_error "Failed to publish message '$header'" >&2
+            return $ret1
+        else
+            log "Published message '$header'"
+        fi
+    fi
+}
+
+
+# Prepare sending of the message.
+#
+# $1 - topic id
+# $2 - title
+# $3 - file
+# global variables:
+#   instance_principal, OCI_CLI_PROFILE_CONFIG, OCI_INSTANCE_PRINCIPAL_AUTH
+send_message() {
+    topic=${1}
+    title=${2}
+    file=${3}
+    #
+    # If oci cli config or api key doesn't exist
+    # switch to instance principal authorization
+    if [ -f "$oci_cli_config_file" -a -f "$oci_api_key_file" ]; then
+        instance_principal=0
+    else
+        instance_principal=1
+    fi
+    #
+    OCI_INSTANCE_PRINCIPAL_AUTH="--auth instance_principal"
+    OCI_CLI_PROFILE_CONFIG="--config-file $oci_cli_config_file"
+    size=$(stat -c %s "$file")
+    #
+    # split file in 64KB chunks if necessary.
+    if [ "$size" -gt 65536 ]; then
+        split -C 64K -d -a 1 "$file" "${file}."
+        files=$(find "$file".*)
+        count=$(echo "$files" | wc -l)
+        echo "$files" | while IFS='' read -r f; do
+            i=${f##*.}
+            t="[$((i+1))/$count] $title"
+            body=$(read_file "$f")
+            publish_message "$topic" "$t" "$body"
+            ret=$?
+            rm -f "$f"
+            # wait a second
+            sleep 1
+        done
+    else
+        body=$(read_file "$file")
+        publish_message "$topic" "$title" "$body"
+        ret=$?
+    fi
+}
+
+#
+# main
+run_as_root_check
```

Comment:
Why doing all these check rather than just add the proper dependencies on the package?